src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_3_horizontal_layout.html.twig

```diff
@@ -25,13 +25,18 @@ col-sm-2
 {# Rows #}
 
 {% block form_row -%}
-    <div class="form-group{% if (not compound or force_error|default(false)) and not valid %} has-error{% endif %}">
-        {{- form_label(form) -}}
-        <div class="{{ block('form_group_class') }}">
-            {{- form_widget(form) -}}
-            {{- form_errors(form) -}}
+    {% set displayMarkup = form.children|length == 0 or form.vars.label is not same as(false) %}
```

Comment:
You should not change the behavior when `form.children|length == 0` imo.
This is not a bug, other templates show the label, and may be part of the work in progress feature in #17609.